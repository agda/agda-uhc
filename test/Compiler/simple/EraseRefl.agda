{-# OPTIONS -v treeless.opt:20 #-}

module _ where

open import Common.Prelude
open import Common.Equality

data Dec {a} (A : Set a) : Set a where
  yes : A → Dec A
  no  : (A → ⊥) → Dec A

decEq₂ : ∀ {a b c} {A : Set a} {B : Set b} {C : Set c} {f : A → B → C} →
           (∀ {x y z w} → f x y ≡ f z w → x ≡ z) →
           (∀ {x y z w} → f x y ≡ f z w → y ≡ w) →
           ∀ {x y z w} → Dec (x ≡ y) → Dec (z ≡ w) → Dec (f x z ≡ f y w)
decEq₂ f-inj₁ f-inj₂ (no neq)    _         = no λ eq → neq (f-inj₁ eq)
decEq₂ f-inj₁ f-inj₂  _         (no neq)   = no λ eq → neq (f-inj₂ eq)
decEq₂ f-inj₁ f-inj₂ (yes refl) (yes refl) = yes refl

main : IO Unit
main = putStrLn ""
