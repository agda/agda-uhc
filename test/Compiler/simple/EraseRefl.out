EXECUTED_PROGRAM

ret > ExitSuccess
out > EraseRefl.decEq₂ =
out >   λ a b c d e f g h i j k l m n o →
out >     case n of
out >       EraseRefl.Dec.yes p →
out >         case o of
out >           EraseRefl.Dec.yes q → EraseRefl.Dec.yes _
out >           EraseRefl.Dec.no q → EraseRefl.Dec.no _
out >       EraseRefl.Dec.no p → EraseRefl.Dec.no _
out > EraseRefl.main = Common.IO.putStrLn ""
out >
out >
